### PR TITLE
public dashboards: move into into its own service

### PR DESCRIPTION
**Pull request by jalevin**
This PR moves public dashboards into it's own service separated from dashboards.
It injects an API into the HTTPServer instead of the public dashboards service which handles mounting routes.

### CODE REVIEW

File: pkg/api/api.go [hunk 1]
```diff
@@ -595,7 +595,7 @@ func (hs *HTTPServer) registerRoutes() {
 	// grafana.net proxy
 	r.Any("/api/gnet/*", reqSignedIn, hs.ProxyGnetRequest)
 
-	// Gravatar service.
+	// Gravatar service.ccess
```
Comment (pianohacker) [inline pkg/api/api.go:598]: Accidental addition

File: pkg/services/publicdashboards/api/api_test.go [hunk 2]
```diff
@@ -0,0 +1,624 @@
...
+
+//    response = callAPI(testServer, http.MethodGet, "/api/public/dashboards/asdf", nil, t)
+//    assert.Equal(t, http.StatusNotFound, response.Code)
+
+//    // control set
+//    testServer = setupTestServer(t, cfg, qs, featuremgmt.WithFeatures(featuremgmt.FlagPublicDashboards), service)
+//    response = callAPI(testServer, http.MethodGet, "/api/public/dashboards/asdf", nil, t)
+//    assert.NotEqual(t, http.StatusNotFound, response.Code)
+//  })
+//}
```
Comment (pianohacker) [inline pkg/services/publicdashboards/api/api_test.go:58]: Should be removed

File: pkg/services/publicdashboards/api/api_test.go [hunk 3]
```diff
@@ -0,0 +1,624 @@
...
+
+	pd, d, err := store.GetPublicDashboard(context.Background(), pubdash.AccessToken)
+	fmt.Println("STORE:", pd)
+	fmt.Println("STORE (dashboard):", d)
+
+	pd2, err := service.GetPublicDashboardConfig(context.Background(), dashboard.OrgId, dashboard.Uid)
+	fmt.Println("STORE:", pd2)
+
+	dash, err := service.GetPublicDashboard(context.Background(), pubdash.AccessToken)
+	fmt.Println("SERVICE (dashboard):", dash)
```
Comment (pianohacker) [inline pkg/services/publicdashboards/api/api_test.go:581]: Debugging artifact, should be removed

File: pkg/services/publicdashboards/api/api_test.go [hunk 4]
```diff
@@ -0,0 +1,624 @@
...
+	server := setupTestServer(t,
+		setting.NewCfg(),
+		qds,
+		featuremgmt.WithFeatures(featuremgmt.FlagPublicDashboards),
+		service,
+		db,
+	)
+
+	pd, d, err = store.GetPublicDashboard(context.Background(), pubdash.AccessToken)
+	fmt.Println("STORE:", pd)
```
Comment (pianohacker) [inline pkg/services/publicdashboards/api/api_test.go:592]: Debugging artifact, should be removed

File: pkg/services/publicdashboards/api/api_test.go [hunk 5]
```diff
@@ -0,0 +1,624 @@
...
+	fmt.Println("STORE (dashboard):", d)
```
Comment (pianohacker) [inline pkg/services/publicdashboards/api/api_test.go:593]: Debugging artifact, should be removed

**Review by pianohacker — DISMISSED** [review]
Looks generally good, just some detritus to get rid of.

**Comment by jalevin** [comment]
Passing onto @pianohacker for merge conflict resolution on Tuesday while I'm OoO

**Comment by jalevin** [comment]
Passing to @pianohacker to get across the line on Tuesday

**Review by owensmallwood — APPROVED** [review]

**Review by timlevett — APPROVED** [review]

**Pull request merged**